enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,444 @@
+`Workers`, resulting in the same default behavior as before this enhancement,
+namely that `IngressController` pod replicas are scheduled on worker nodes and
+determined according to the `InfrastructureTopology`.
+
+In the future, when IPI-installed single control-plane node clusters in the
+cloud no longer provision a load-balancer by default, they would also benefit
+from having the installer set the `IngressPlacement` to `ControlPlane`.
+
+### Risks and Mitigations
+
+This should make no noticable difference on "regular" single control-plane node
+clusters which do not have any day-1 or day-2 worker nodes. The only difference
+for those clusters would be the `IngressController` targeting the "master" pool
+rather than the "worker" pool, but since the single control-plane node is
+already both in the "master" and "worker" pools, that should make no practical
+difference.
+
+I do not believe this enhancement has any security implications.
+
+## Design Details
+
+### Open Questions
+
+None that I can think of at the moment
+
+### Test Plan
+
+- Add unit tests to the Cluster Ingress Operator to make sure the
+IngressController resource is generated as exepected.
+
+- Add periodic nightly tests which install single-node in the cloud, add a few
+worker nodes to it, then run conformance tests to make sure we don't run into
+any problems not described in this enhancement.
+
+- Add periodic nightly tests which install a single-node "none"-platform cluster,
+add worker nodes to it, and check that ingress traffic still works as expected
+and recovers even after the `router-default` pod gets deleted and rescheduled.
+Make sure this is still true even after upgrades.
+
+- Add tests on both cloud / "none"-platform that check that a single-node cluster
+with additional workers recovers after the single control-plane node reboots by
+running conformance tests post-reboot.
+
+- Add tests on both cloud / "none"-platform that check that a single-node cluster
+with additional workers recovers after an upgrade by running conformance tests
+post-upgrade.
+
+TODO: Describe day-1-workers tests?
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+In the non-goals section it's mentioned that this enhancement does not apply to
+clusters which have been installed prior to the enhancement, so their upgrade
+is not discussed.
+
+This enhancement, to the best of my knowledge, should have no problems
+persisting across cluster upgrades. The Test Plan section describes how this
+will be tested.
+
+### Version Skew Strategy
+
+Does not apply, to the best of my understanding.
+
+### Operational Aspects of API Extensions
+
+- The "ControlPlane" IngressPlacement parameter value may not be used if the cluster's
+ControlPlaneTopology parameter is set to "External". 
```

Comment:
Specifying `ingressPlacement: ControlPlane` with `controlPlaneTopology: External` is an interesting combination.  Maybe we can leave that as reserved for possible future use.

Reply:
Not sure I understand what you mean by "Leaving it as reserved"? For now I'll say that this combination is forbidden, I don't think it will be a "Breaking change" to say otherwise in the future

Reply:
Yeah, that's what I mean.  It might be useful for, say, some Hypershift use-case, so I'd rather say it is invalid now but not necessarily invalid forever.